.github/workflows/go.yml

```diff
@@ -0,0 +1,186 @@
+name: Go CI
+
+on:
+    workflow_dispatch:
+    push:
+        branches: [ "main" ]
+        paths:
+            - glide-core/**
+            - submodules/**
+            - go/**
+            - .github/workflows/go.yml
+    pull_request:
+        paths:
+            - glide-core/**
+            - submodules/**
+            - go/**
+            - .github/workflows/go.yml
+
+# Run only the latest job on a branch and cancel previous ones
+concurrency:
+    group: ${{ github.head_ref || github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    build-and-test-go-client:
+        timeout-minutes: 10
+        strategy:
+            # Run all jobs
+            fail-fast: false
+            matrix:
+                go:
+                    - '1.18'
+                    - '1.21'
+                redis:
```

Comment:
@barshaul we should probably try to export the redis/os matrices to a shared configuration, instead of repeating them everywhere.